spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourcePoolFilterSpecTest.java

```diff
@@ -0,0 +1,254 @@
+                ProductAttributes.Cores.withValue("4")
+            );
+            ProductDTO product3 = createProductByOwner(adminClient, owner, Products.random());
+
+            createPools(adminClient, owner, Pools.random(product1),
+                Pools.random(product2), Pools.random(product3));
+        }
+
+        @Test
+        public void shouldAllowOwnersFilterPoolsBySingleFilter() throws ApiException {
+            List<PoolDTO> filteredPools = adminClient.owners().listOwnerPoolsWithAttributes(
+                owner.getKey(), List.of("support_level:VIP"));
+            assertEquals(1, filteredPools.size());
+            assertEquals(product1.getId(), filteredPools.get(0).getProductId());
+
+            //Now with wildcards
+            filteredPools = adminClient.owners().listOwnerPoolsWithAttributes(
+                owner.getKey(), List.of("support_level:V*P"));
+            assertEquals(filteredPools.size(), 1);
+            assertEquals(product1.getId(), filteredPools.get(0).getProductId());
+
+            filteredPools = adminClient.owners().listOwnerPoolsWithAttributes(
+                owner.getKey(), List.of("support_level:V?P"));
+            assertEquals(filteredPools.size(), 1);
+            assertEquals(product1.getId(), filteredPools.get(0).getProductId());
+        }
+
+        @Test
+        public void shouldAllowOwnersFilterPoolsByMultipleFilter() throws ApiException {
+            List<PoolDTO> filteredPools = adminClient.owners().listOwnerPoolsWithAttributes(
+                owner.getKey(), List.of("support_level:Supurb", "cores:4"));
+            assertEquals(1, filteredPools.size());
+            assertEquals(product2.getId(), filteredPools.get(0).getProductId());
+        }
+
+        @Test
+        public void shouldNotAllowOwnersFilterPoolsWithBadFormatAttribute() throws ApiException {
+            OwnerDTO owner = createOwner(adminClient);
+
+            // Try to send attribute with bad splitter(should be colon)
+            assertBadRequest(() -> {
+                adminClient.owners().listOwnerPoolsWithAttributes(
+                    owner.getKey(), List.of("support_level;Supurb"));
+            });
+        }
+
+        @Test
+        public void shouldAllowOwnersFilterPoolsByPoolId() throws ApiException {
+            List<PoolDTO> pools = adminClient.owners().listOwnerPools(owner.getKey());
+            for (PoolDTO pool : pools) {
+                List<PoolDTO> filteredPools = adminClient.owners().listOwnerPoolsWithPoolIds(
+                    owner.getKey(), List.of(pool.getId()));
+                assertThat(filteredPools)
+                    .map(PoolDTO::getId)
+                    .containsOnly(pool.getId());
+            }
+        }
+
+        @Test
+        public void shouldAllowOwnersFilterPoolsByMultiplePoolId() throws ApiException {
+            List<PoolDTO> pools = adminClient.owners().listOwnerPools(owner.getKey());
```

Comment:
We should add an assertion on this to verify that this output contains at least two elements, as the remainder of the test is reliant upon that.

Reply:
done